### DISCUSSION TITLE
Convert script to importable package

### DISCUSSION THREAD

**Discussion by ppfeister** [FlareSolverr/FlareSolverr#1293] [Ideas]
FlareSolverr is typically dockerized, but it can be ran as a script if need be. The problem is, this is a pretty poor way to integrate it into other projects.

Just as an example, I'm experimenting with using FlareSolverr as a part of [my newer project here][oculus flare branch]. To do so, I have to include FlareSolverr's repository as a submodule and then do some nonsense to treat it as properly importable. Namespace packaging has issues with FlareSolverr as well, so an \_\_init\_\_ file is often needed.

Is there any interest in or opposition to converting FlareSolverr to a properly importable package? This would allow for it to be included in other projects much more easily.

I recently converted [Sherlock][sherlock] to a package as well, which allowed for it's easier inclusion in the previously linked project (and others) just the same.

[oculus flare branch]: https://github.com/ppfeister/oculus/tree/feature/flare
[sherlock]: https://github.com/sherlock-project/sherlock

**Comment by ppfeister** [comment]
@ilike2burnthing I went down a rabbit hole and converted FlareSolverr over in https://github.com/ppfeister/FlareSolverr/tree/refactor/pkg.

Assuming you have Chrome/Chromium and `xorg-x11-server-Xvfb` (or your distrib's equiv), give it a shot:

```bash
pip install flaresolverr
python -m flaresolverr
```

Should start right up. Much easier to integrate into other projects now. Assuming there's interest, the Dockerfile and similar would just need to be tweaked before bringing any changes forward as a pr.

**Reply by ppfeister** [reply]
Oh, right --- this version also removes undetected-chromedriver from the flaresolverr repo, and instead includes it as a dependency from pypi. No reason to statically embed anymore now that it's available there.

**Reply by ilike2burnthing** [reply]
What about the FlareSolverr-specific changes we had made to UC?

**Reply by ppfeister** [reply]
Let me investigate further. There didn't seem to be much in the way of changes but it sounds like I just botched my diff somehow

Depending on the severity of these changes, I wonder if it makes more sense to overload/overwrite or just keep it embedded.... hm. Probably just embedded as it was before.

Will update shortly

**Comment by ilike2burnthing** [comment]
Some things have been reverted or changed, but basically anything that wasn't a version bump was specific to us - https://github.com/FlareSolverr/FlareSolverr/commits/master/src/undetected_chromedriver

**Comment by ppfeister** [comment]
FS's custom UC has been returned as a submodule

New rc/demo pushed to pypi under the same name

**Comment by ppfeister** [comment]
rc{2..4} pushed with Drission to reflect https://github.com/FlareSolverr/FlareSolverr/pull/1300#issuecomment-2259586944
